scripts/articleView.js

```diff
@@ -29,48 +22,94 @@ articleView.populateFilters = function() {
+          return $(this).find('.byline a').text() == $('#author-filter').val();
+      }).show();
 
     } else {
-      // TODO: If the select box was changed to an option that is blank, we should
-      //       show all the articles, except the one article we are using as a template.
+      // DONE: If the select box was changed to an option that is blank, we should show all the articles, except the one article we are using as a template.
+
+      $('#articles article').each(function() {
+          $(this).show();
+      });
 
     }
     $('#category-filter').val('');
   });
 };
 
 articleView.handleCategoryFilter = function() {
-  // TODO: Just like we do for #author-filter above, we should handle change events on the #category-filter element.
-  //       When an option with a value is selected, hide all the articles, then reveal the matches.
-  //       When the blank (default) option is selected, show all the articles, except for the template.
-  //       Be sure to reset the #author-filter while you are at it!
+  // DONE: Just like we do for #author-filter above, we should handle change events on the #category-filter element. When an option with a value is selected, hide all the articles, then reveal the matches. When the blank (default) option is selected, show all the articles, except for the template. Be sure to reset the #author-filter while you are at it!
+  $('#category-filter').on('change', function() {
+    if ($(this).val()) {
+
+      $('#articles article').each(function() {
+          $(this).hide();
+      });
+
+      $('#articles article').filter(function() {
+          return $(this).attr("data-category") == $("#category-filter").val();
+      }).show();
+
+    } else {
+
+      $('#articles article').each(function() {
+          $(this).show();
+      });
+
+    }
+    $('#author-filter').val('');
+  });
 
 };
 
 articleView.handleMainNav = function() {
-  // TODO: Add an event handler to .main-nav element that will power the Tabs feature.
-  //       Clicking any .tab element should hide all the .tab-content sections, and then reveal the
-  //       single .tab-content section that is associated with the clicked .tab element.
-  //       So: You need to dynamically build a selector string with the correct ID, based on the
-  //       data available to you on the .tab element that was clicked.
-  $('.main-nav').on(/* CODE GOES HERE */);
-
-  $('.main-nav .tab:first').click(); // Let's now trigger a click on the first .tab element, to set up the page.
+  // DONE: Add an event handler to .main-nav element that will power the Tabs feature. Clicking any .tab element should hide all the .tab-content sections, and then reveal the single .tab-content section that is associated with the clicked .tab element. So: You need to dynamically build a selector string with the correct ID, based on the data available to you on the .tab element that was clicked.
+  $('.main-nav').on("click", function() {
```

Comment:
we can pass .tab as a parameter here and then run a if else statement based on our different tabs instead of adding two event listeners inside of this